pytorch_lightning/loops/epoch/evaluation_epoch_loop.py

```diff
@@ -134,10 +133,13 @@ def advance(
         self.trainer.logger_connector.update_eval_step_metrics()
 
         # track epoch level outputs
-        if self._should_track_batch_outputs_for_epoch_end():
-            output = recursive_detach(output, to_cpu=self.trainer.move_metrics_to_cpu)
-            if output is not None:
-                self.outputs.append(output)
+        if self._should_track_batch_outputs_for_epoch_end() and output is not None:
+            self.outputs.append(output)
+
+        if self.trainer.move_metrics_to_cpu:
+            # the evaluation step output is not moved as they are not considered "metrics"
+            assert self.trainer._results is not None
+            self.trainer._results.cpu()
```

Comment:
This would break if there is reduction performed on epoch end.

Reply:
You mean as described in https://github.com/PyTorchLightning/pytorch-lightning/issues/10595 right?

It's something to address separately, this PR just makes it works exactly as for training